### DISCUSSION TITLE
Search term modifiers should override default behavior

### DISCUSSION THREAD

**Discussion by mattstein** [craftcms/cms#10613] [Ideas] [answered]
Right now, the [defaultSearchTermOptions](https://craftcms.com/docs/3.x/config/config-settings.html#defaultsearchtermoptions) `subLeft` and `subRight`, when `true`, basically add an implicit `*` to the beginning or end of the search string. As a result, using a search keyword that includes `*` may not have any effect because those defaults are still applied regardless of the content of the search keywords(s) used.

Ideally, specific modifiers in the search string should take precedence over the defaults.

## Example

Let’s say we have stock Craft settings and the following entry titles to be searched:

- `Royalty`
- `Type`
- `Style`

### Current Behavior

| Results Searching `ty` | Results Searching `*ty` |
| - | - |
| `Royalty`<br>`Type`<br>`Style` | `Royalty`<br>`Type`<br>`Style`

### Suggested Behavior

| Results Searching `ty` | Results Searching `*ty` |
| - | - |
| `Royalty`<br>`Type`<br>`Style` | `Royalty`

**Comment by brandonkelly** [comment, marked as answer]
Changed for the next Craft 4 release! (8c3ba35e02e621569c7ba17519cc99a38db3b83d)

**Reply by brandonkelly** [reply]
Craft 4.0.0-beta.4 is out now with that change.
